### DISCUSSION TITLE
How to add two URLs for one emulator

### DISCUSSION THREAD

**Discussion by steve1977** [RandomNinjaAtk/docker-raromprocessor#21] [Q&A]
Need a bit of coding help... I tried to add two URLs for one emulator, but failed. I tried both adding it as separate entries or as two links under the same entry. Any chance help advice me on this? Below what I have tried so far.

**Attempt 1 - Added two URLs**
	if echo "$folder" | grep "^gc$" | read; then
		ConsoleId=16
		ConsoleName="GameCube"
		ConsoleDirectory="gc"
		ArchiveUrl="$(curl -s "https://archive.org/download/GCRedumpNKitPart1/" | grep ".nkit.gcz" | grep -io '<a href=['"'"'"][^"'"'"']*['"'"'"]' |   sed -e 's/^<a href=["'"'"']//i' -e 's/["'"'"']$//i' | sed 's/\///g' | sort -u | sed 's|^|https://archive.org/download/GCRedumpNKitPart1/|')"
		keepCompressed=true
		ArchiveUrl="$(curl -s "https://archive.org/download/GCRedumpNKitPart2/" | grep ".nkit.gcz" | grep -io '<a href=['"'"'"][^"'"'"']*['"'"'"]' |   sed -e 's/^<a href=["'"'"']//i' -e 's/["'"'"']$//i' | sed 's/\///g' | sort -u | sed 's|^|https://archive.org/download/GCRedumpNKitPart2/|')"
		keepCompressed=true
	fi

**Attempt 2 - Two entries**
	if echo "$folder" | grep "^gc$" | read; then
		ConsoleId=16
		ConsoleName="GameCube"
		ConsoleDirectory="gc"
		ArchiveUrl="$(curl -s "https://archive.org/download/GCRedumpNKitPart1/" | grep ".nkit.gcz" | grep -io '<a href=['"'"'"][^"'"'"']*['"'"'"]' |   sed -e 's/^<a href=["'"'"']//i' -e 's/["'"'"']$//i' | sed 's/\///g' | sort -u | sed 's|^|https://archive.org/download/GCRedumpNKitPart1/|')"
		keepCompressed=true
	fi
	if echo "$folder" | grep "^gc$" | read; then
		ConsoleId=16
		ConsoleName="GameCube"
		ConsoleDirectory="gc"
		ArchiveUrl="$(curl -s "https://archive.org/download/GCRedumpNKitPart2/" | grep ".nkit.gcz" | grep -io '<a href=['"'"'"][^"'"'"']*['"'"'"]' |   sed -e 's/^<a href=["'"'"']//i' -e 's/["'"'"']$//i' | sed 's/\///g' | sort -u | sed 's|^|https://archive.org/download/GCRedumpNKitPart2/|')"
		keepCompressed=true
	fi

**Comment by RandomNinjaAtk** [comment]
The current implementation/design is not designed to handle multiple URLs per system, you would have to likely write a loop for the download process to support multiple urls.

**Comment by steve1977** [comment]
Got it. I got Gamecube running now (for one part). Will close previous pull request (which was still wrong) and open a new one, which then is correct & tested.

Any chance you can help me with the loop for multiple URL and I could then add the link for the second URL?

**Comment by steve1977** [comment]
Below two pull requests are ready to merge. And would appreciate your help with the loop. I will then also work on Wii and PS2 support, which is still under testing:

https://github.com/RandomNinjaAtk/docker-raromprocessor/pull/9
https://github.com/RandomNinjaAtk/docker-raromprocessor/pull/22
